lib/utils/ignore.ts

```diff
@@ -200,6 +200,41 @@ export interface FileStats {
 	stats: Stats;
 }
 
+/**
+ * Create a list of files for the filesystem subtree rooted at
+ * projectDir, excluding entries for directories themselves.
+ * @param projectDir Source directory (root of subtree to be listed)
+ */
+async function listFiles(projectDir: string): Promise<string[]> {
+	const dirs: string[] = [];
+	const files: string[] = [];
+	dirs.push(projectDir);
+	async function walk(currentDirs: string[]): Promise<string[]> {
+		if (!currentDirs.length) {
+			return files;
+		}
+
+		const foundDirs: string[] = [];
+
+		await Promise.all(
+			currentDirs.map(async (dir) => {
```

Comment:
If you don't mind checking, what would the speed impact be (in seconds) if we did **not** use `Promise.all` + `.map` to parallelise the calls to `fs.readdir`? I.e., some less "aggressive" coding like:
```suggestion
		for (const dir of currentDirs) {
```

And also adding the `await` keyword to `return await walk(foundDirs);` (see other comment too).

I suspect the aggressive parallelisation could be blamed for the memory exhaustion of `listFiles` a year ago. Does this parallelisation mean that `fs.readdir` would run simultaneously for potentially **ALL** directories in the project at once? E.g. 100,000 directories (exaggeration) would mean running `fs.readdir` 100,000 times in parallel with simultaneous stack memory use for 100,000 calls to `listFiles`. Or maybe I am misunderstanding it!

I gather that `recursive-fs` does **not** do aggressive parallelisation thanks to the following test in callback style:

https://github.com/simov/recursive-fs/blob/v2.1.0/lib/read.js#L29-L30
```js
        if (++complete === _dirs.length) {
          walk(__dirs)
        }
```

Reply:
@pdcastro This is an await/async version of what recursive-fs does, which prevents recursion into subdirectories until the current level has been scanned. I haven't tested a linear/non-async approach but based on what I've seen from other implementations I'd expect it to be about 30% slower.
The memory usage is almost certainly down to unnecessarily calling `stat` AFAICT.